### DISCUSSION TITLE
Uploading the code…

### DISCUSSION THREAD

**Discussion by plingboot** [DCC-EX/EX-T3-WiFi#73] [Q&A]
Just wondered whether anyone could offer assistance with the upload issues I'm having.
I'm on a MacPro, with the FireBeetle connected to one of the USB3 ports.
I've copied the contents of the SD folder to the micro SDCard and 'stumbled' my way through installing Xcode, VSC and PlatformIO.
The list of available options down the side of PIO are slightly different to the ones on the tutorial page - not sure if that's a problem.
Everything seems to be working correctly (when clicking upload) until the point where it's supposed to connect to the Firebeetle.
(see attached pic). 

![Screenshot 2023-03-13 at 10 15 35](https://user-images.githubusercontent.com/452962/224673723-dca45186-9241-497f-9650-1c4236e4a864.png)

**Comment by alex-code** [comment]
I'm not familiar with macs.
Is there a way to check if the FireBeetle is detected and available on a COM port?

**Reply by plingboot** [reply]
Yeah, just checked and my Mac doesn't seem to be seeing it - I've been through a few troubleshooting YouTube clips and loaded various drivers, but it's not working. Looks like I'll need to try this on a PC later…

**Reply by alex-code** [reply]
Not sure if the info here will help, https://learn.sparkfun.com/tutorials/how-to-install-ch340-drivers/all#mac-osx
I'm on W11 and the driver must've been pre-installed as I didn't have to do anything extra.

**Reply by plingboot** [reply]
Thanks for that !! Not sure why the driver I downloaded didn't work, but after a couple of attempts that installed and I've been able to upload to the FireBeetle. Looking forward to giving it a test later today. 
PS nice bit of component packaging with the enclosure. I resin printed it and it looks really good.

**Reply by alex-code** [reply]
Glad you got it working.
Thanks for the feedback, I'd liked to have made it slightly slimmer but needed to give room for easy wiring and not knowing what size batteries users would be able to get.

I'd like to see pics of that, I looked into resin printing but think it was over like 12 hours to finish!
